providers/dns/ibmcloud/internal/wrapper.go

```diff
@@ -66,7 +67,27 @@ func getDomainID(service services.Dns_Domain, domain string) (*int, error) {
 		return r.Id, nil
 	}
 
-	return nil, fmt.Errorf("no data found of domain: %s", domain)
+	// The domain was not found by name. For subdomains this is not unusal in softlayer.
+	// So in case a subdomain like sub.toplevel.domain was used try again using the
+	// parent domain (strip the first part in the domain string -> toplevel.domain).
```

Comment:
```suggestion
	// The domain was not found by name.
	// For subdomains this is not unusual in softlayer.
	// So in case a subdomain like `sub.toplevel.tld` was used try again using the parent domain
	// (strip the first part in the domain string -> `toplevel.tld`).
```